### DISCUSSION TITLE
help with input registration order in RHF

### DISCUSSION THREAD

**Discussion by spector163** [react-hook-form/react-hook-form#10134] [Q&A] [answered]
Inputs are not registered in order that they are in the render function of the form. Controlled inputs are placed last even if they are first in the render function. Codesandbox for the issue [https://ewoet3.csb.app/](https://ewoet3.csb.app/) which in turn causes a weird focus issue where if an uncontrolled input is errored while submitting it will get focus before the controlled one even if the controlled component is first in the order.

**Comment by leapful** [comment, marked as answer]
@spector163 

You are using `Controller` component from react-hook-form which always register field name on its `render` prop so even that order of `Controller` is first element, it always be registered after other `input` elements that are being rendered directly with `register` method.

You can perform `register` for field name related to `Controller` component just like normal input element to achieve correct order as a workaround.

Working demo: https://codesandbox.io/s/gracious-varahamihira-itl250?file=/src/App.tsx
